File: Quadratic/C#/Quadratic equation/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Quadratic_equation
{
    public partial class Form1 : Form
    {
        Double A;
        Double B;
        Double C;
        Double D;
        Double X1;
        Double X2;
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (Double.TryParse(textBoxA.Text, out A))
            {
                if (Double.TryParse(textBoxB.Text, out B))
                {
                    if (Double.TryParse(textBoxC.Text, out C))
                    {
                        D = Math.Pow(B, 2) - 4 * A * C;
                        textBoxD.Text = D + "";
                        if (D < 0)
                        {
                            textBoxX1.Text = "Error";
                            textBoxX2.Text = "D<0, Can't solve";
                        }
                        else if (D == 0)
                        {
                            X1 = -B / 2 * A;
                            textBoxX1.Text = X1 + "";
                            textBoxX2.Text = "";
                        }
                        else if (D>0)
                        {
                            X1 = (-B + Math.Sqrt(D)) / (2 * A);
                            X2 = (-B - Math.Sqrt(D)) / (2 * A);
                            textBoxX1.Text = X1 + "";
                            textBoxX2.Text = X2 + "";                       
                        }
                    }
                }
             }
        }
    }
}

File: Quadratic/VisualBasic/Quadratic equation/Form1.vb
Public Class Form1
    Dim A As Decimal
    Dim B As Decimal
    Dim C As Decimal
    Dim D As Decimal
    Dim X1 As Decimal
    Dim X2 As Decimal

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If IsNumeric(TextBoxA.Text) = True Then
            A = TextBoxA.Text
            If IsNumeric(TextBoxB.Text) = True Then
                B = TextBoxB.Text
                If IsNumeric(TextBoxC.Text) = True Then
                    C = TextBoxC.Text
                End If
            End If
        End If
        D = B ^ 2 - 4 * A * C
        TextBoxD.Text = D

        If D < 0 Then
            TextBoxX1.Text = "Error"
            TextBoxX2.Text = "D<0, cant solve"
        ElseIf D = 0 Then
            X1 = -B / 2 * A
            TextBoxX1.Text = X1
            TextBoxX2.Text = ""
        Else
            X1 = (-B + Math.Sqrt(D)) / (2 * A)
            X2 = (-B - Math.Sqrt(D)) / (2 * A)
            TextBoxX1.Text = X1
            TextBoxX2.Text = X2
        End If
    End Sub
End Class
